### PR TITLE
Shop item info

### PR DESCRIPTION
**Pull request by rjdellecese**
No PR message is safe.

### CODE REVIEW

File: src/Application/Components/Shop/KeyBoundShopItem.jsx [hunk 1]
```diff
@@ -16,6 +18,10 @@ class KeyBoundShopItem extends React.Component {
 
   render() {
     const imgSrc = `images/${this.props.name}.png`;
+    let itemDetails = false;
```
Comment (samgqroberts) [inline src/Application/Components/Shop/KeyBoundShopItem.jsx]: `{undefined}` and `{false}` are equivalent in the html block so u only need `let itemDetails;`

File: src/Application/Components/Shop/KeyBoundShopItem.jsx [hunk 2]
```diff
@@ -16,6 +18,10 @@ class KeyBoundShopItem extends React.Component {
...
+    if (this.props.areStatsVisible) {
+      itemDetails = <ShopItemStats key={this.props.name} name={this.props.name} />;
```
Comment (samgqroberts) [inline src/Application/Components/Shop/KeyBoundShopItem.jsx]: when using the same prop (or just accessing `this.props` a bunch) u can use this notation ```const { areStatsVisible, name, owned } = this.props

File: src/Application/Components/Shop/KeyBoundShopItem.jsx [hunk 3]
```diff
@@ -27,12 +33,15 @@ class KeyBoundShopItem extends React.Component {
           </div>
           <div className="shop-item-number col-md-2">{this.props.owned}</div>
         </div>
+
```
Comment (samgqroberts) [inline src/Application/Components/Shop/KeyBoundShopItem.jsx]: no newlines in html block, the `</div>` lines serve to visually chop it up well enough

File: src/Application/Components/Shop/Shop.jsx [hunk 4]
```diff
@@ -1,20 +1,25 @@
...
 
   render() {
     return (
         <div className="panel panel-primary">
 
-          <div className="panel-heading">
-            <h3 className="panel-title broker-title text-center">Real Estate</h3>
+          <div className="panel-heading text-center">
+            <h3 className="panel-title broker-title">Real Estate</h3>
+            <br />
```
Comment (samgqroberts) [inline src/Application/Components/Shop/Shop.jsx]: i strongly prefer to do line breaks as part of the css but meh i won't die on that hill.  with css / styling a lot of the time it's like "well if it looks right"

File: src/Application/Components/Shop/Shop.jsx [hunk 5]
```diff
@@ -27,10 +32,12 @@ const Shop = React.createClass({
                   owned={asset.owned}
                   keyCode={assetDefaults[asset.name].keyCode}
                   onClick={_.partial(this.props.onItemClick, asset.id)}
+                  areStatsVisible={this.props.areStatsVisible}
                 />
               );
             })}
           </div>
+
```
Comment (samgqroberts) [inline src/Application/Components/Shop/Shop.jsx]: ah i see there's newlines all over the place in aaron's existing code.  if you want to chop up a long html block more, extract chunks of it out to their own `renderSomething()` methods

File: src/Application/Models/Broker.js [hunk 6]
```diff
@@ -51,7 +56,7 @@ class Broker {
     const index = _.findIndex(this._assets, { id: assetId });
     if (index === -1) {
       // Return the original broker, but this is not good.
-      return new Broker(this._assets);
+      return new Broker(this._assets, this._areStatsVisible);
```
Comment (samgqroberts) [inline src/Application/Models/Broker.js]: why not just `return this;` ?  seems to be functionally equivalent but saving the re-construction

File: src/Application/Models/Broker.js [hunk 7]
```diff
@@ -84,12 +89,17 @@ class Broker {
...
     }
     const newAsset = this._assets[index].unlock();
 
     return this._insertIntoNewBroker(newAsset, index);
   }
+
+  toggleStatsVisibility() {
+    const areStatsVisible = this._areStatsVisible ? false : true;
```
Comment (samgqroberts) [inline src/Application/Models/Broker.js]: can simplify to `const areStatsVisible = !this._areStatsVisible;`

File: src/Application/Models/Broker.js (hunk 6 above)
Comment (rjdellecese) [inline src/Application/Models/Broker.js]: Yup yup yup gonna do that.

File: src/Application/Components/Shop/Shop.jsx (hunk 5 above)
Comment (rjdellecese) [inline src/Application/Components/Shop/Shop.jsx]: Shall update in Shop and shall keep a lookout for this in other components.

File: src/Application/Components/Shop/KeyBoundShopItem.jsx [hunk 8]
```diff
@@ -15,29 +19,42 @@ class KeyBoundShopItem extends React.Component {
   }
 
   render() {
-    const imgSrc = `images/${this.props.name}.png`;
+    const imgSrc = `images/${this.props.asset.name}.png`;
+    let itemDetails = false;
```
Comment (samgqroberts) [inline src/Application/Components/Shop/KeyBoundShopItem.jsx]: `let itemDetails;`

File: src/Application/Components/Shop/KeyBoundShopItem.jsx (hunk 2 above)
Comment (rjdellecese) [inline src/Application/Components/Shop/KeyBoundShopItem.jsx]: That's pretty dope

**Pull request merged**